freelist_test.go

```diff
@@ -44,6 +44,131 @@ func TestFreelist_release(t *testing.T) {
 	}
 }
 
+type testRange struct {
```

Comment:
why do you define these types out of the func body? do you plan to reuse them in other tests?

Reply:
I'll move them into the test function. Thanks!